src/execute_linter.cc

```diff
@@ -24,24 +25,166 @@
 #include "absl/strings/string_view.h"
 #include "src/lint_errors.h"
 #include "src/linter.h"
+#include "src/linter_options.h"
 #include "zetasql/base/status.h"
 #include "zetasql/base/status_macros.h"
 
 namespace zetasql::linter {
 
 namespace {
+
+// This function will parse "NOLINT"(<CheckName>) syntax from
+// a single line of comment.
+// If NOLINT usage errors occur, it counts as a lint
+// error and it will be returned in a result.
+LinterResult ParseNoLintSingleComment(absl::string_view line,
+                                      LinterOptions* options, int position) {
+  LinterResult result;
+  std::map<std::string, ErrorCode> error_map = GetErrorMap();
+  int i = 0;
+
+  // Skip spaces from the beginning.
+  while (i < static_cast<int>(line.size()) && line[i] == ' ') ++i;
+
+  // Found enabling or disabling type of comment.
+  if (line.substr(i, 6) == "NOLINT" || line.substr(i, 4) == "LINT") {
+    // Stores if it is "NOLINT" or "LINT"
+    bool is_it_disabling = (line.substr(i, 6) == "NOLINT");
+    while (i < static_cast<int>(line.size()) && line[i] != '(') ++i;
+    ++i;
+
+    std::string check_name = "";
+    while (i < static_cast<int>(line.size()) && line[i - 1] != ')') {
```

Comment:
Please consider using regular expression to to find matches in a the given string (Google’s current regular expression library is called RE2: https://github.com/google/re2)
 
check http://www.regular-expressions.info or the interactive tutorial at http://regexone.com.

E.g:

const LazyRE2 kRegex = { R"((LINT|NOLINT)\(([A-Za-z0-9]*)\))" };
string check_name;
string lint_comment;
bool matched = RE2::FullMatch(s, *kRegex, &lint_comment, &check_name);

Reply:
I figured out it is a liitle more complcated. I changed the code according to regex, but we lost the ability to detect column number.
a line comment can be:
 NOLINT  ( alias, djdjdjdj, \t parser-failed ) \n
It is saying column number in this implementation, but I couldn't figure out how to do it in regex.